firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorViewController.swift

```diff
@@ -0,0 +1,205 @@
+            descriptionLabel.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            descriptionLabel.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            descriptionLabel.topAnchor.constraint(
+                equalTo: header.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            passwordField.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            passwordField.trailingAnchor.constraint(equalTo: contentView.trailingAnchor),
+            passwordField.topAnchor.constraint(
+                equalTo: descriptionLabel.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            usePasswordButton.leadingAnchor.constraint(equalTo: contentView.leadingAnchor),
+            usePasswordButton.trailingAnchor.constraint(
+                equalTo: passwordField.trailingAnchor),
+            usePasswordButton.topAnchor.constraint(
+                equalTo: passwordField.bottomAnchor,
+                constant: UX.containerElementsVerticalPadding),
+            usePasswordButton.bottomAnchor.constraint(equalTo: contentView.bottomAnchor)
+        ])
+    }
+
+    // MARK: - Themable
+    func applyTheme() {
+        let theme = themeManager.getCurrentTheme(for: windowUUID)
+        view.backgroundColor = theme.colors.layer1
+        descriptionLabel.textColor = theme.colors.textSecondary
+        usePasswordButton.applyTheme(theme: theme)
+        passwordField.applyTheme(theme: theme)
+        header.applyTheme(theme: theme)
+    }
+
+    private func buildUsePasswordButton() {
+        let usePasswordButtonVM = PrimaryRoundedButtonViewModel(
+            title: .PasswordGenerator.UsePasswordButtonLabel,
+            a11yIdentifier: AccessibilityIdentifiers.PasswordGenerator.usePasswordButton)
+        usePasswordButton.configure(viewModel: usePasswordButtonVM)
+    }
+
+    // MARK: - Redux
+    func subscribeToRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.showScreen,
+                screen: .passwordGenerator
+            )
+        )
+
+        let uuid = windowUUID
+        store.subscribe(self, transform: {
+            return $0.select({ appState in
+                return PasswordGeneratorState(appState: appState, uuid: uuid)
+            })
+        })
+    }
+
+    func unsubscribeFromRedux() {
+        store.dispatch(
+            ScreenAction(
+                windowUUID: windowUUID,
+                actionType: ScreenActionType.closeScreen,
+                screen: .passwordGenerator
+            )
+        )
+    }
+
+    func newState(state: PasswordGeneratorState) {
+        passwordGeneratorState = state
+        passwordField.configure(password: passwordGeneratorState.password)
+    }
+
+    // MARK: - Notifiable
+    private func applyDynamicFontChange() {
+        header.applyDynamicFontChange()
```

Comment:
Is there a reason you update the header and passwordField here when the dynamic type changes? Did you consider setting up the notification in the views itself?